### PR TITLE
CSR report to Socrata ETL

### PR DESCRIPTION
**Pull request by Charlie-Henry**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/12776
https://github.com/cityofaustin/atd-data-tech/issues/12379

## Associated repo
https://github.com/cityofaustin/atd-executive-dashboard

## Testing

**Steps to test:**
1. Checkout this branch
2. Trigger this DAG either through the UI or `docker compose run --rm airflow-cli dags test atd_executive_csr_data`
3. Check logs to see if Socrata is getting new data  similar to:`{'Errors': 0, 'Rows Created': 0, 'Rows Updated': 1050, 'Rows Deleted': 0}`

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Add note to 1PW secrets moved to API vault and check for duplicates

### CODE REVIEW

File: dags/atd_executive_csr_data.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    catchup=False,
+) as dag:
+    env_vars = get_env_vars_task(REQUIRED_SECRETS)
+
+    t1 = DockerOperator(
+        task_id="csr_report_to_socrata",
+        image=docker_image,
+        api_version="auto",
+        auto_remove=True,
+        command=f"python csr/csr_to_socrata.py",
```
Comment (mddilley) [inline dags/atd_executive_csr_data.py:107]: 👏 for [this warning](https://github.com/cityofaustin/atd-executive-dashboard/blob/6eabdfe7bc0510cd58a6231a228ce8149ecb95fd/csr/csr_to_socrata.py#L58). I wasn't on the VPN initially, and this had me running in no time.

**Review by mddilley — APPROVED** [review]
Tested by triggering in my local dashboard, and it ran successfully. 🚢

File: dags/atd_executive_csr_data.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        "opitem": "Microstrategy API",
+        "opfield": "shared.Password",
+    },
+}
+
+with DAG(
+    dag_id="atd_executive_csr_data",
+    description="Downloads a report of 311 service requests for TPW and publishes it in a Socrata dataset.",
+    default_args=default_args,
+    schedule_interval="36 9 * * *" if DEPLOYMENT_ENVIRONMENT == "production" else None,
```
Comment (frankhereford) [inline dags/atd_executive_csr_data.py:95]: 9:36 AM - is there a reason to pick this arbitrary time? It's as good as any other, but it just stuck out as curious.

**Review by frankhereford — APPROVED** [review]
This works great Charlie. Thank you for this. 

In the output, I see this pandas warning a few times: 

```
INFO - /usr/local/lib/python3.8/site-packages/pandas/core/frame.py:4441: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame
See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
```

They are referring to [these assignments](https://github.com/cityofaustin/atd-executive-dashboard/blob/main/csr/csr_to_socrata.py#L136-L142):

```
    df["state_plane_x_coordinate"] = df["state_plane_x_coordinate"].replace(
        {np.nan: None}
    )
    df["state_plane_y_coordinate"] = df["state_plane_y_coordinate"].replace(
        {np.nan: None}
    )
    df["sr_location"] = df["sr_location"].replace({np.nan: None})
```

In this case, I don't think this is a major deal because this chained assignments don't need to be a single operation that will succeed or fail as a unit and the nesting of the data frames isn't deep. I thought you might want to see it though.

Thanks again! Please feel free to ask for reviews on your ETLs if you want; I'm always interested in trying to keep up with all the new stuff you're cranking out. 🚀

File: dags/atd_executive_csr_data.py (hunk 2 above)
Comment (Charlie-Henry) [reply dags/atd_executive_csr_data.py:95]: Just a random time I picked, but I believe the report we get from 3-1-1 is run daily at 8AM.

**Comment by johnclary** [comment]
yayyyy it's live!

**Pull request merged**